src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -751,7 +762,10 @@ export class SearchResult extends Disposable {
 
 	private _onChange = this._register(new Emitter<IChangeEvent>());
 	readonly onChange: Event<IChangeEvent> = this._onChange.event;
-
+	private _onRefreshSearchResult = this._register(new PauseableEmitter<IChangeEvent>({
```

Comment:
Why add this event instead of using onChange?

Reply:
I wanted to isolate the pause to just refreshing the tree and have all other onChange events fire as usual.

Reply:
What else is onChange used for? I don't really think that's necessary.